### PR TITLE
added support for .mjs module resolution

### PR DESCRIPTION
**Pull request by manizm**
Fixes Issue #15 by supporting `.mjs` modules

### CODE REVIEW

**Comment by manizm** [comment]
@ahocevar let me know if this is good enough
It simply makes sure that if `.mjs or .js` extension is present, use that, otherwise default back to `.js` if no extension is provided

File: index.js [hunk 1]
```diff
@@ -20,24 +20,23 @@ if (!fs.existsSync(moduleRootAbsolute)) {
 const importRegEx = /import\(["']([^"']*)["']\)\.([^ \.\|\}><,\)=#\n]*)([ \.\|\}><,\)=#\n])/g;
 const typedefRegEx = /@typedef \{[^\}]*\} (\S+)/g;
 const noClassdescRegEx = /@(typedef|module|type)/;
+const extensionReplaceRegEx = /\.js|.mjs?$/;
```
Comment (ahocevar) [inline index.js]:
```suggestion
const extensionReplaceRegEx = /\.m?js$/;
```

File: index.js [hunk 2]
```diff
@@ -20,24 +20,23 @@ if (!fs.existsSync(moduleRootAbsolute)) {
...
 const slashRegEx = /\\/g;
 
 const moduleInfos = {};
 const fileNodes = {};
 
-function getModuleInfo(moduleId, parser) {
+function getModuleInfo(moduleId, extension = '', parser) {
```
Comment (ahocevar) [inline index.js]:
```suggestion
function getModuleInfo(moduleId, extension, parser) {
```

**Review by ahocevar — CHANGES_REQUESTED** [review]
Thanks for your work on this, @manizm.

Instead of calculating the extension in several places, I'd suggest adding a helper function: 
```js
getExtension(absolutePath) {
  return extensionReplaceRegEx.test(absolutePath) ? absolutePath.match(extensionReplaceRegEx)[0] . '.js';
}
```

**Review by ahocevar — APPROVED** [review]
Thanks, @manizm

**Pull request merged**